src/main/java/org/osc/controller/nsc/restserver/api/InspectionPortApis.java

```diff
@@ -0,0 +1,177 @@
+@Produces({ MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML })
+public class InspectionPortApis {
+    private static Logger LOG = Logger.getLogger(InspectionPortApis.class);
+    private EntityManager em;
+    private RedirectionApiUtils utils;
+    private SampleSdnRedirectionApi sdnApi;
+    @Reference(target = "(osgi.local.enabled=true)")
+    private TransactionControl txControl;
+    @Reference(target = "(osgi.unit.name=nsc-mgr)")
+    private EntityManagerFactoryBuilder builder;
+    @Reference(target = "(osgi.jdbc.driver.class=org.h2.Driver)")
+    private DataSourceFactory jdbcFactory;
+    @Reference(target = "(osgi.local.enabled=true)")
+    private JPAEntityManagerProviderFactory resourceFactory;
+    private String id;
+
+    public void init(String id) throws Exception {
+        if (id != null) {
+            if (id.equals(this.id)) {
+                return;
+            }
+        } else {
+            return;
+        }
+
+        Properties props = new Properties();
+        props.setProperty(JDBC_URL, "jdbc:h2:./nscPlugin_" + id + ";MVCC\\=TRUE;LOCK_TIMEOUT\\=10000;MV_STORE=FALSE;");
+        props.setProperty(JDBC_USER, "admin");
+        props.setProperty(JDBC_PASSWORD, "admin123");
+        DataSource ds = null;
+        try {
+            ds = this.jdbcFactory.createDataSource(props);
+        } catch (SQLException error) {
+            LOG.error(error);
+            throw new IllegalStateException(error.getMessage(), error);
+        }
+        EntityManager em = this.resourceFactory
+                .getProviderFor(this.builder, singletonMap("javax.persistence.nonJtaDataSource", (Object) ds), null)
+                .getResource(this.txControl);
+        this.em = em;
+        this.sdnApi = new SampleSdnRedirectionApi(this.txControl, this.em);
+        this.utils = new RedirectionApiUtils(this.em, this.txControl);
+        this.id = id;
+
+    }
+
+    @Path("/controller/{id}/insport")
+    @POST
+    public String createInspectionPort(@PathParam("id") String id, InspectionPortEntity entity)
+            throws Exception {
+        if (entity == null) {
+            throw new IllegalArgumentException("Attempt to create null InspectionPort");
+        }
+        LOG.info(String.format("Creating inspection port for (ingress id %s ; egress id %s)",
+                entity.getIngressPort().getElementId(), entity.getEgressPort().getElementId()));
+        init(id);
+        InspectionPortEntity inspectionPort = (InspectionPortEntity) this.sdnApi.registerInspectionPort(entity);
+        return inspectionPort == null ? null : inspectionPort.getElementId();
+    }
+
+    @Path("/controller/{id}/insport/{portId}")
+    @PUT
+    public InspectionPortEntity updateInspectionPort(@PathParam("id") String id,
+            @PathParam("portId") String portId, InspectionPortEntity entity) throws Exception {
+
+        LOG.info(String.format("Updating the inspection port element id %s ", portId));
```

Comment:
Similar comment here, you could also move this code as a method into `SampleSdnRedirectionApi`